pkgs/build-support/flatten-references-graph/src/flatten_references_graph/split_paths.py

```diff
@@ -0,0 +1,226 @@
+from toolz import curried as tlz
+from toolz import curry
+
+from .lib import (
+    debug,
+    debug_plot,
+    DEBUG_PLOT,
+    find_vertex_by_name_or_none,
+    graph_is_empty,
+    is_None,
+    subcomponent_multi,
+    unnest_iterable
+)
+
+
+@curry
+def add_root(added_root_name, graph):
+    """Add single root to the graph connected to all existing roots.
+
+    If graph has only one root, return the graph unchanged and the name
+    of the root vertex.
+
+    Otherwise return a modified graph (copy) and a name of the added root
+    vertex.
+    """
+    roots = graph.vs.select(lambda v: len(graph.predecessors(v)) == 0)
+    root_names = roots["name"]
+
+    if len(root_names) == 1:
+        return graph, root_names[0]
+    else:
+        edges = [(added_root_name, v) for v in root_names]
+        graph_with_root = graph + added_root_name + edges
+        return graph_with_root, added_root_name
+
+
+@curry
+def remove_added_root(added_root_name, graph):
+    """Remove vertex with given name, returning copy of input graph if vertex
+    with given name is found in the graph
+    """
+    added_root = find_vertex_by_name_or_none(graph)(added_root_name)
```

Comment:
The name tells me that it'd remove an added root, but it actually removes any vertex. Why does it have `added_root` in the name?

Reply:
how about move `added_root_name` to top level, rename it to `fake_root_name`, like so:

```python
fake_root_name = "__root__"
```

and then call this function `remove_fake_root`?

as this function goes in tandem with `add_root` (to be renamed: `coerce_to_singly_rooted_graph`) the `added_root_name` param would be removed there as well